### ISSUE TITLE
"Embed Pods Frameworks" build phase on extensions without frameworks causes rejection by Apple on submission

### ISSUE THREAD

**Issue by TheDom** [CocoaPods/CocoaPods#4383]
We recently upgraded from version 0.35 to 0.39 RC 1. With that update `pod install` started to add a "Embed Pods Frameworks" build phase to our Today and Share extensions. However, we're still using static libraries instead of frameworks.

When we tried to submit our app to Apple with Xcode 7.0.1, it was rejected with the following errors:

```
ERROR ITMS-90206: "Invalid Bundle. The bundle at 'MyApp.app/PlugIns/MyAppShare.appex' contains disallowed file 'Frameworks'."
ERROR ITMS-90206: "Invalid Bundle. The bundle at 'MyApp.app/PlugIns/MyAppToday.appex' contains disallowed file 'Frameworks'."
```

<img width="631" alt="xcode_-_itms-90206_png_2_402x846_pixels" src="https://cloud.githubusercontent.com/assets/204015/10464332/1b0d5cac-719e-11e5-94fd-7c6580f90851.png">

This was not fixed by simply setting "Embedded Content Contains Swift Code" to YES for our app like various Stack Overflow answers suggested – e.g. http://stackoverflow.com/questions/25777958/validation-error-invalid-bundle-the-bundle-at-contains-disallowed-file-fr

The problem was actually caused by an empty `Frameworks` folder inside our our two `.appex` packages. Removing the two build phases manually fixed the problem.

However, as said earlier, `pod install` always re-adds the build phases which is unnecessary and causes said problem.

**Comment by segiddins** [comment]
There are many open issues for this already. In the meantime, you can add a post-install hook to your podfile to remove the build phase.

**Comment by ChristianAtEmoJam** [comment]
Hi, the problem removing the build phase is that the app fails App Store submission.
I recently submitted an app with a 3rd party keyboard ; I removed the Embed Pods Frameworks script in the Build Phases for my extension ; the archive was submitted successfully, but the validation team rejected the app because they could not install the keyboard.
It seems the trick helps passing the submission, but not the validation, and I can not publish my app on the App Store... 

Regards

**Comment by seamane** [comment]
I am having the same problem. I can archive it successfully, but I always receive the archive error given above. I have tried changing all relevant build settings to use the $(inherited) flag, but that hasn't worked. I have also deleted the empty Framework directory, but still no luck.

I am using pod --version 0.39.0 and XCode version 7.0.1.

Also when I try to update with 'pod install', I get a warning that my "target overrides the 'OTHER_CFLAGS' build setting" defined in the pods. I have replaced everything with the inherited flag, but still can't get rid of this warning. I don't know if this has to do with submission problem, but I thought I would let you know.

**Comment by simX** [comment]
Is this issue fixed?  What's the proper resolution?  Is this only applicable to extensions?  Only to projects with Swift?

**Comment by segiddins** [comment]
@simX as the issue is still open, it's not fixed yet.

**Comment by woodydaniel** [comment]
Also experiencing this issue.

**Comment by Nadohs** [comment]
I am getting the same warning in when building with cocoapods 0.39.0. my project included a custom keyboard extension, which does not have any dependancies. Also getting this error
```
... /Debug-iphonesimulator/TargetName.app/TargetName

ld: framework not found Pods
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by neonichu** [comment]
I think this has been fixed by #4203 in the meantime.
